src/cmap/connection_pool.ts

```diff
@@ -493,25 +492,16 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
   private interruptInUseConnections(minGeneration: number) {
     for (const connection of this[kCheckedOut]) {
       if (connection.generation <= minGeneration) {
-        this.checkIn(connection);
         connection.onError(new PoolClearedOnNetworkError(this));
+        this.checkIn(connection);
```

Comment:
@baileympearson this was done to fix a test failure where we were getting an uncaught error when running [this spec test](https://github.com/mongodb/specifications/blob/345c243748390f17072754a0b07bc2dd4580b8c9/source/server-discovery-and-monitoring/tests/unified/interruptInUse-pool-clear.yml#L232)

Reply:
Yeah, this is right.  Good catch.

I think the reason this worked in the past was because

1. `checkIn` calls `destroyConnection()`, which calls `connection.destroy()` in a process.nextTick()
2. then `connection.onError() runs and shuts down the connection 
3. then `connection.destroy()` is a no-op because the connection has already been destroyed

You made `destroyConnection` synchronous, so we need to run `onError` first to ensure the connection is closed with the correct error.  Then we run checkIn, which resets the pool state and calls `connection.destroy()`, which is a no-op.